R/reactives.R

```diff
@@ -400,7 +404,12 @@ Observable <- R6Class(
       on.exit(.running <<- wasRunning)
 
       ctx$run(function() {
-        result <- withVisible(try(shinyCallingHandlers(.func()), silent=TRUE))
+        result <- withCallingHandlers(withVisible(.func()),
+          error = function(cond) {
+            .visible <<- FALSE
+            .value <<- try(stop(cond), silent = TRUE)
```

Comment:
Yep, I'll add a comment. It'd be even better to have a separate flag (like `.visible`) tracking whether it's an error and just storing the plain condition. But this is fine for now.